Event Documentation Start

VapourDrive opened the pull request:
I began work on documenting events, with an overview, a general warning about the nature of events for novice users and a fleshed out example. With the start to finish registry example.

This is by no means a fully comprehensive covering of events, but a good introduction to them.
I am interested in providing more event documentation based on what is wanted.

matthewprenger commented on events/usingevents.md:
```diff
@@ -0,0 +1,64 @@
+
+The basics for subscribing to an event are as follows:
+
+'''
```
Wrong characters. You should be using the ` instead of ' for code blocks

VapourDrive commented:
You commented just as I realized the misshap, sorry about that. fixing now.

matthewprenger commented:
@VapourDrive Minor nitpick. You can use:
 \`\`\`java
    class Foo {}
 \`\`\`
to get code coloring.

VapourDrive commented:
That does look a lot nicer! I'll remember that for next time.

cpw commented:
Can you please correct some spelling mistakes? Wether is whether throughout the document. Sorry, I just want to make sure the little things don't bug me..

cpw commented:
@VapourDrive why the sudden close attempt? This looks ok, but it needs a little bit of tidying up is all..

VapourDrive commented:
I can do make the spelling mistake changes, I'll scan the document for any other issues.

VapourDrive commented:
I have fixed the two instances of "wether" and added the paths to the the mkdocs file. 
@cpw can you confirm that the mkdocs file is what is desired? And maybe provide feedback on the content and style.

cpw commented on events/overview.md:
```diff
@@ -0,0 +1,14 @@
+
+The event system in forge/fml provides a way for a mod to change something that happens in the game without breaking down compatibility.
+
+An event is void function that gets fired when "an event" happens in the game and gives mods the opportunity to change the outcome by subscribing to the event.
```
Is a void function?

cpw commented on events/usingevents.md:
````diff
@@ -0,0 +1,64 @@
+}
+```
+
+The event priority that is **optional** to set determines what order the subscribers get acces to the event
````
access

cpw commented on events/usingevents.md:
````diff
@@ -0,0 +1,64 @@
+```
+
+The event priority that is **optional** to set determines what order the subscribers get acces to the event
+If you have the lowest priority you get access last, but you get the last chance to set what happens in the event.
````
called, not accessed. The priority affects the calling sequence for dispatching events to the handlers.

VapourDrive replied:
Fair enough. Proper terminology is key for good documentation.

cpw commented on events/usingevents.md:
````diff
@@ -0,0 +1,64 @@
+}
+```
+
+The newSpeed float is the only variable that can actually be set set, as all the others are final. Here we simply add one float to the original speed if the player has greater than 0 levels .
````
In this example, the newSpeed float is the only factor that can be changed. In this case, we are adding one to it, increasing the break speed, for an experienced player.

VapourDrive replied:
This is probably a more eloquent, and less ambiguous way of describing what is happening.

cpw commented on events/usingevents.md:
````diff
@@ -0,0 +1,64 @@
+```
+
+The newSpeed float is the only variable that can actually be set set, as all the others are final. Here we simply add one float to the original speed if the player has greater than 0 levels .
+The event can be canceled.
````
This specific event can be cancelled, causing the break to not start.

cpw commented on events/eventnature.md:
```diff
@@ -0,0 +1,11 @@
+===========
+
+When you want to do something with an event, you register your subscription, and you set up your block of code that gets run.
+(More on the specifics elsewhere)
```
Where?

VapourDrive replied:
I'll reference the example that I made.

VapourDrive commented:
I'll try to change what needs changing as soon as possible.

LexManos commented:
Poke for update.
